tests/integration/conftest.py

```diff
@@ -357,6 +411,31 @@ async def app_juju_storage(
     return application
 
 
+@pytest_asyncio.fixture(scope="module", name="test_github_branch")
+async def test_github_branch_fixture(github_repository: Repository) -> AsyncIterator[Branch]:
+    """Create a new branch for testing, from latest commit in current branch."""
+    test_branch = f"test-{secrets.token_hex(4)}"
+    branch_ref = github_repository.create_git_ref(
+        ref=f"refs/heads/{test_branch}", sha=Repo().head.commit.hexsha
+    )
+
+    def get_branch():
+        """Get newly created branch."""
+        try:
+            branch = github_repository.get_branch(test_branch)
+        except GithubException as err:
+            if err.status == 404:
```

Comment:
Why this special treatment if not found?

Reply:
There's a lag between branch creation and getting the branch from GitHub, so essentially we're trying a few more times :)